unified_planning/model/action.py

```diff
@@ -226,7 +226,7 @@ def add_decrease_effect(self, fluent: Union['up.model.fnode.FNode', 'up.model.fl
         self._add_effect_instance(up.model.effect.Effect(fluent_exp, value_exp, condition_exp, kind = up.model.effect.EffectKind.DECREASE))
 
     def _add_effect_instance(self, effect: 'up.model.effect.Effect'):
-        if effect not in self._effects:
+        if _check_conflicting_or_existing_effects(effect, self._simulated_effect, self._effects):
```

Comment:
here you reuse `self._effects`, saving memory but the cost is that each effect addition becomes a linear-complexity operation....